Codecademy/gamut
# feat(ToolTip): New + Improved

### Overview


New ToolTips + InfoTips, including adding required InfoTips to IconButtons


### PR Checklist

- [x] Related to designs: [here](https://www.figma.com/file/ReGfRNillGABAj5SlITalN/branch/ayKNSg6QvZUjsgw0FFysW4/%F0%9F%93%90-Gamut?type=design&node-id=41538-55277&mode=design&t=MtL6PFyyDUGSIcAm-0)
- [x] Related to JIRA ticket: [GM-595]

#### Testing Instructions


Check out the new Tip pages + look thru documentation

- Navigate to InfoTip (for both `floating` + `inline` placements) 
    - Click one of the InfoTips - the tip should show
    - Click it again or click outside - the tip should hide
    - The same should happen on VO and it should be narrated to the user
- Navigate to ToolTip (for both `floating` + `inline` placements) 
    - focus or hover over one of the buttons - the tip should show
    - unhover or unfocus - the tip should hide
    - on VO - the button should be labeled by it’s associated ToolTip
- Check out Alert page
    - See the ToolTip on hover of the close button
- Check out GridForm page -> navigate to the ‘With InfoTip’ section
    - The InfoTips should work as outlined above
- Check out Dialog, Modal, + Flyout
    -   See the close buttons all have tooltips

#### PR Links and Envs

| Repository   | PR Link                                                  | PR Env                                                   |
| :----------- | :------------------------------------------------------- | :------------------------------------------------------- |
| Monolith     | [Monolith PR](https://github.com/codecademy-engineering/Codecademy/pull/37384)  | [Monolith Env](http://www.google.fr/ 'Named link title') |
| Portal       | [Portal Link](https://github.com/codecademy-engineering/mono/actions/runs/8381414012/job/22952940398?pr=5212)  | [Portal Env](https://tengu.codecademy.com/catalog?PR_ENV=portal-app-pr-5212)   |



[GM-595]: https://codecademy.atlassian.net/browse/GM-595?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
